test/start.js

```diff
@@ -0,0 +1,96 @@
+const exec = require('child_process').exec;
+const fs = require('fs-extra');
+const path = require('path');
+
+const strapiBin = path.resolve('./packages/strapi/bin/strapi.js');
+const appName = 'testApp';
+let appStart;
+
+const databases = {
+  mongo: `--dbclient=mongo --dbhost=127.0.0.1 --dbport=27017 --dbname=strapi-test-${new Date().getTime()} --dbusername="" --dbpassword=""`,
+  postgres: `--dbclient=postgres --dbhost=127.0.0.1 --dbport=5432 --dbname=strapi-test --dbusername="" --dbpassword=""`,
+  mysql: `--dbclient=mysql --dbhost=127.0.0.1 --dbport=3306 --dbname=strapi-test --dbusername="root" --dbpassword="root"`
+};
+
+const {runCLI: jest} = require('jest-cli/build/cli');
+
+const main = async () => {
+  const clean = () => {
+    return new Promise((resolve) => {
+      fs.exists(appName, exists => {
+        if (exists) {
+          fs.removeSync(appName);
+        }
+
+        resolve();
+      });
+    });
+  };
+
+  const generate = (database) => {
+    return new Promise((resolve, reject) => {
+      const appCreation = exec(
+        `node ${strapiBin} new ${appName} --dev ${database}`,
+      );
+
+      appCreation.stdout.on('data', data => {
```

Comment:
It's better to use this event to detect if the server started correctly or not. If the event is not precise enough, we should improve it https://github.com/strapi/strapi/blob/master/packages/strapi/lib/Strapi.js#L123

Reply:
Sorry, apply this comment to the next method